util/disk-image-validator/run_config_tests.py

```diff
@@ -0,0 +1,161 @@
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import os
+import subprocess
+
+"""
+gem5 Configuration Test Script
+
+This script runs a specified workload across multiple gem5 configurations to verify
+that gem5 can successfully boot full-system workloads with different CPU models
+and cache hierarchies.
+
+Usage:
+    python3 run_config_tests.py --workload <WORKLOAD_ID> --resource_version <RESOURCE_VERSION>
+
+Arguments:
+    --workload           The ID of the workload to be executed.
+    --resource_version   The version of the workload resource.
+
+Example:
+    python3 run_config_tests.py --workload linux-boot --resource_version 1.0
+
+Requirements:
+    - gem5 binary must be built and located at: ./build/ALL/gem5.opt
+    - Configuration script must be present at: util/disk-image-validator/config_tester.py
+    - Python 3 must be installed
+    - Required Python modules: argparse, subprocess, os
+
+Functionality:
+    - Runs gem5 with different configurations:
+        * KVM_test
+        * MESI_cache_test
+        * O3_test
+        * MINOR_test
+        * ATOMIC_test
+    - Outputs results in m5out/<workload_id>-<test_name>
+    - Prints a summary of test results and exits with an appropriate status code.
+
+Exit Codes:
+    - 0: All tests passed.
+    - 1: One or more tests failed.
+"""
+
+
+# Path to gem5 binary
+GEM5_BINARY = "./build/ALL/gem5.opt"
+
+# Path to the config script
+GEM5_CONFIG = "util/disk-image-validator/config_tester.py"
+
+# List of configuration tests to run
+CONFIG_TESTS = [
+    "KVM_test",
+    "MESI_cache_test",
+    "O3_test",
+    "MINOR_test",
+    "ATOMIC_test",
+]
```

Comment:
You're missing testing multicore setups. I'd create a test for single-core, 2,4, and 8-core setups